gammapy/scripts/make_bg_cube_models.py

```diff
@@ -0,0 +1,308 @@
+from ..background import make_bg_cube_model
+# TODO: revise imports!!!
+
+__all__ = ['make_bg_cube_models',
+           'create_bg_observation_list',
+           'group_observations',
+           'stack_observations',
+           ]
+
+
+DEBUG = 1 # 0: no output, 1: output, 2: NOTHING, 3: more verbose
+# TODO: remove the DEBUG global variable, when the logger works!!!
+
+def main(args=None):
+    parser = get_parser(make_bg_cube_models)
+    parser.add_argument('fitspath', type=str,
+                        help='Dir path to input event list fits files.')
+    parser.add_argument('--test', type=bool, default=False,
+                        help='If true, use a subset of observations '
+                        'for testing purposes')
+    parser.add_argument("-l", "--loglevel", default='info',
+                        choices=['debug', 'info', 'warning', 'error', 'critical'],
+                        help="Set the logging level")
+    args = parser.parse_args(args)
+
+    set_up_logging_from_args(args)
+
+    make_bg_cube_models(**vars(args))
+
+
+def make_bg_cube_models(fitspath,
+                        test):
+    """Create background cube models from the complete dataset of an experiment.
+
+    Starting with gamma-ray event lists and effective area IRFs,
+    make background templates. Steps
+
+    1. make a global event list from a datastore
+    2. filter the runs keeping only the ones far from known sources
+    3. group the runs according to similar observation conditions (i.e. alt, az)
+        * using `~gammapy.obs.ObservationGroups`
+    4. create a bg cube model for each group using:
+        * the `~gammapy.background.make_bg_cube_model` method
+        * and `~gammapy.background.CubeBackgroundModel` objects as containers
+
+    The models are stored into FITS files.
+
+    It can take a few minutes to run. For a quicker test, please activate the
+    **test** flag.
+
+    TODO: revise doc!!!
+
+    Parameters
+    ----------
+    fitspath : str
+        Path to dir containing event list fits files and a list of them.
+    """
+    create_bg_observation_list(fitspath, test)
+    group_observations(test)
+    stack_observations(fitspath)
+
+
+def create_bg_observation_list(fits_path, test):
+    """Make total observation list and filter the observations.
+
+    In a first version, all obs taken within 3 deg of a known source
+    will be rejected. If a source is extended, twice the extension is
+    added to the corresponding exclusion region radius of 3 deg.
+
+    TODO: on a second version, one could only filter out the runs
+          too close to the galacic plane, and afterwards use masks
+          to cut out sources, for runs taken on extragalactic objects
+          (correcting the livetime accordingly).
+
+    TODO: move function to background/obs module? But where?!!!
+
+    Parameters
+    ----------
+    fits_path : str
+        path to dir containing list of input fits event files
+    """
+    if DEBUG:
```

Comment:
Just to be clear: I said before that you should use logging statements everywhere you have print statements.
But actually most of your print statements, like this, should just be removed IMO.

Please use your best judgement (maybe err on the side of leaving too many debug / info / warning statement) and then I'll leave inline comments which ones I think are useful to keep.